en/02_Developer_Guides/00_Model/04_Data_Types_and_Casting.md

```diff
@@ -31,33 +31,41 @@ class Player extends DataObject
-- `'Int'`: A 32-bit signed integer field (see: [DBInt](api:SilverStripe\ORM\FieldType\DBInt)).
-- `'Locale'`: A field for storing locales (see: [DBLocale](api:SilverStripe\ORM\FieldType\DBLocale)).
-- `'Money'`: Similar to Currency, but with localisation support (see: [DBMoney](api:SilverStripe\ORM\FieldType\DBMoney)).
-- `'MultiEnum'`: An enumeration set of strings that can store multiple values (see: [DBMultiEnum](api:SilverStripe\ORM\FieldType\DBMultiEnum)).
-- `'Percentage'`: A decimal number between 0 and 1 that represents a percentage (see: [DBPercentage](api:SilverStripe\ORM\FieldType\DBPercentage)).
-- `'PolymorphicForeignKey'`: A special ForeignKey class that handles relations with arbitrary class types (see: [DBPolymorphicForeignKey](api:SilverStripe\ORM\FieldType\DBPolymorphicForeignKey)).
-- `'PrimaryKey'`: A special type Int field used for primary keys. (see: [DBPrimaryKey](api:SilverStripe\ORM\FieldType\DBPrimaryKey)).
-- `'Text'`: A variable-length string of up to 2MB, designed to store raw text (see: [DBText](api:SilverStripe\ORM\FieldType\DBText)).
-- `'Time'`: A time field (see: [DBTime](api:SilverStripe\ORM\FieldType\DBTime)).
-- `'Varchar'`: A variable-length string of up to 255 characters, designed to store raw text (see: [DBVarchar](api:SilverStripe\ORM\FieldType\DBVarchar)).
-- `'Year'`: Represents a single year field (see: [DBYear](api:SilverStripe\ORM\FieldType\DBYear)).
+| Field | Description | Validation | API Reference |
+| --- | --- | --- | --- |
+| `BigInt` | An 8-byte signed integer field | Must be an int between -9223372036854775808 and 9223372036854775807 | [`DBBigInt`](api:SilverStripe\ORM\FieldType\DBBigInt) |
+| `Boolean` | A boolean field stored as a tinyint | Must be a boolean | [`DBBoolean`](api:SilverStripe\ORM\FieldType\DBBoolean) |
+| `Currency` | A number with 2 decimal points of precision, designed to store currency values | Must be a decimal | [`DBCurrency`](api:SilverStripe\ORM\FieldType\DBCurrency) |
+| `Date` | A date field | Must be a valid date in `Y-m-d` format | [`DBDate`](api:SilverStripe\ORM\FieldType\DBDate) |
+| `Datetime` | A date/time field | Must be a valid datetime in `Y-m-d H:i:s` format | [`DBDatetime`](api:SilverStripe\ORM\FieldType\DBDatetime) |
+| `DBClassName` | A special enumeration for storing class names | Must be a valid subclass name | [`DBClassName`](api:SilverStripe\ORM\FieldType\DBClassName) |
+| `DBClassNameVarchar` | A special enumeration for storing class names in a `Varchar` field | Must be a valid subclass name | [`DBClassNameVarchar`](api:SilverStripe\ORM\FieldType\DBClassNameVarchar) |
+| `Decimal` | A decimal number | Must be a decimal. | [`DBDecimal`](api:SilverStripe\ORM\FieldType\DBDecimal) |
+| `Double` | A floating point number with double precision | Not validated | [`DBDouble`](api:SilverStripe\ORM\FieldType\DBDouble) |
```

Comment:
We don't validate it at all? Please use a relevant validator. Even just straight numeric validator would be better than nothing.

Reply:
Looks like there are some others below that also lack validation. Please give them appropriate validation instead.

Reply:
Have added NumericFieldValidator to DBFloat + DBDouble, will redo kitchen sink run shortly

DBText, DBHTMLText and DBHTMLFragment actually already has StringFieldValidator on them, have updated docs